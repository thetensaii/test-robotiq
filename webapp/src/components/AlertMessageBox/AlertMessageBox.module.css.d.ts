declare const styles: {
  readonly "alert": string;
  readonly "alertDanger": string;
  readonly "alertSuccess": string;
  readonly "alertInfo": string;
  readonly "closeButton": string;
};
export = styles;

