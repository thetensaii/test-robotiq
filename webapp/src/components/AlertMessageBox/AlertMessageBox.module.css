.alert {
	position: fixed;
	top: 30px;

	left: 50%;
	transform: translate(-50%, -50%);
	padding: 10px 20px;
	border-radius: 15px;
	z-index: 10;
	box-shadow: 0 7px 14px rgba(0, 0, 0, 0.25), 0 5px 5px rgba(0, 0, 0, 0.22);

	display: flex;
	align-items: center;
	justify-content: space-around;
}

.alertDanger {
	color: white;
	background-color: #de2100;
}

.alertSuccess {
	color: black;
	background-color: #2bff4b;
}

.alertInfo {
	color: black;
	background-color: #2bdfff;
}

.closeButton {
	all: unset;
	display: flex;
	align-items: center;
	justify-content: center;

	margin-left: 15px;
}

.closeButton:hover {
	cursor: pointer;
}
