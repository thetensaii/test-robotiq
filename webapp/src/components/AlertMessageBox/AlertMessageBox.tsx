import React from 'react';
import { createPortal } from 'react-dom';
import styles from './AlertMessageBox.module.css';
import { MdClose } from 'react-icons/md';

export enum AlertMessageBoxTypes {
	danger = 'alertDanger',
	info = 'alertInfo',
	success = 'alertSuccess',
}

type AlertMessageBoxProps = {
	type?: AlertMessageBoxTypes;
	closeAlert: () => void;
};

export const AlertMessageBox: React.FC<AlertMessageBoxProps> = ({
	type = AlertMessageBoxTypes.info,
	closeAlert,
	children,
}) => {
	return createPortal(
		<div className={`${styles.alert} ${styles[type]}`}>
			{children}
			<button className={styles.closeButton} onClick={closeAlert}>
				<MdClose />
			</button>
		</div>,
		document.getElementById('alert') as Element
	);
};
