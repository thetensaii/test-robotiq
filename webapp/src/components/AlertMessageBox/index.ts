export { AlertMessageBox as default } from './AlertMessageBox';
export * from './AlertMessageBox';
