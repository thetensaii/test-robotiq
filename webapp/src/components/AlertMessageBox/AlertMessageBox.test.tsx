import React from 'react';
import { render, screen } from '@testing-library/react';
import { AlertMessageBox } from './AlertMessageBox';
import userEvent from '@testing-library/user-event';

describe('AlertMessageBox component tests', () => {
	it('should renders AlertMessageBox', () => {
		const closeAlert = jest.fn();

		const alertContainer = document.createElement('div');
		alertContainer.setAttribute('id', 'alert');

		const { container } = render(
			<AlertMessageBox closeAlert={closeAlert}>Alert test</AlertMessageBox>,
			{
				container: document.body.appendChild(alertContainer),
			}
		);
		expect(container).toMatchSnapshot();
	});

	it('should renders message', () => {
		const closeAlert = jest.fn();

		const alertContainer = document.createElement('div');
		alertContainer.setAttribute('id', 'alert');

		render(<AlertMessageBox closeAlert={closeAlert}>Alert test</AlertMessageBox>, {
			container: document.body.appendChild(alertContainer),
		});

		const messageElement = screen.getByText(/alert test/i);

		expect(messageElement).toBeInTheDocument();
	});

	it('should close message box', async () => {
		const closeAlert = jest.fn();
		const user = userEvent.setup();

		const alertContainer = document.createElement('div');
		alertContainer.setAttribute('id', 'alert');

		render(<AlertMessageBox closeAlert={closeAlert}>Alert test</AlertMessageBox>, {
			container: document.body.appendChild(alertContainer),
		});

		const buttonElement = screen.getByRole('button');

		expect(buttonElement).toBeInTheDocument();
		await user.click(buttonElement);

		expect(closeAlert).toHaveBeenCalledTimes(1);
	});
});
