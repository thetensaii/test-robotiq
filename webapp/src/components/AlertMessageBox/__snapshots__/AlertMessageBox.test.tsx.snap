// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertMessageBox component tests should renders AlertMessageBox 1`] = `
<div
  id="alert"
>
  <div
    class="alert alertInfo"
  >
    Alert test
    <button
      class="closeButton"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </button>
  </div>
</div>
`;
